layouts/shortcodes/api-reference-link.html

```diff
@@ -0,0 +1,41 @@
+<!-- Check if 'api-metadata' exists in the front-matter of the file -->
+{{- if .Page.Params.api_metadata -}}
+<!-- Extract API metadata -->
+{{- $apiVersion := .Page.Params.api_metadata.apiVersion -}}
+{{- $kind := .Page.Params.api_metadata.kind -}}
+{{- $type := .Page.Params.api_metadata.type -}}
+{{- $version := $apiVersion -}}
+
+<!-- Define variable to store API reference links -->
+{{- $apiReferenceLinks := "" -}}
+{{- $apiReferenceText := T "api_reference_title" -}}
+
+<!-- Get all files under the specified directory -->
+{{- $apiRefDir := printf "docs/reference/kubernetes-api/%s-resources" $type -}}
+{{- $apiReferenceFiles := site.GetPage $apiRefDir -}}
+
+<!-- Loop through API reference files -->
+{{- range $apiRefFile := $apiReferenceFiles.RegularPages -}}
+  {{- if and (ne .Section "") (in .File.Dir $apiRefDir) -}}
+    <!-- Check if the file's metadata matches -->
+    {{- with .Params.api_metadata -}}
+      {{- if and (eq .kind $kind) (eq .apiVersion $version) -}}
+        <!-- If the file's metadata matches, add link to variable -->
+        {{- $link := printf "<a id='api-reference-page-link' href='%s'>%s %s</a>" $apiRefFile.Permalink $kind $apiReferenceText -}}
```

Comment:
Maybe use a `class` not `id`. It's possible that one page might link to two APIs; for example, https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/ references CertificateSigningRequest _and_ ClusterTrustBundle.